agent/bench-scripts/pbench-linpack

```diff
@@ -108,48 +120,131 @@ while true; do
 benchmark_iterations="${benchmark_run_dir}/.iterations"
 
 mkdir -p ${benchmark_run_dir}/.running
 
 # now that the benchmark_run_dir directory exists, we can initialize the iterations file
 > ${benchmark_iterations}
 
-iteration=1-$threads-threads
-sample="sample1"
-echo $iteration >> $benchmark_iterations
-benchmark_results_dir="$benchmark_run_dir/$iteration/${sample}"
-result_file=$benchmark_results_dir/result.txt
-mkdir -p $benchmark_results_dir
-
 # Start the tool meisters on each registered local/remote host
+echo "Starting tool-meister"
 pbench-tool-meister-start --sysinfo="${sysinfo}" "${tool_group}"
 if [[ ${?} != 0 ]]; then
-	error_log "[${script_name}]: failed to start the tool meisters."
-	exit 1
+    error_log "[${script_name}]: failed to start the tool meisters."
+    exit 1
 fi
 
 trap "interrupt" INT QUIT TERM
 
-## Run the benchmark and start/stop perf analysis tools
-pbench-start-tools --group=$tool_group --dir=$benchmark_results_dir
-
-OMP_NUM_THREADS=$threads ${linpack_dir}/${linpack_cmd} < ${linpack_dir}/${linpack_dat} | tee $result_file
-
-pbench-stop-tools --group=$tool_group --dir=$benchmark_results_dir
-pbench-send-tools --group=$tool_group --dir=$benchmark_results_dir
-pbench-postprocess-tools --group=$tool_group --dir=$benchmark_results_dir
-
-ln -s ${sample} $benchmark_run_dir/$iteration/reference-result
+count=1
+for thread in ${threads//,/ }; do
+    iteration="$count-$thread-threads"
+    echo $iteration >> $benchmark_iterations
+    for sample in `seq 1 $nr_samples`; do
+        echo "Starting sample $iteration/sample$sample"
+        benchmark_results_dir="$benchmark_run_dir/$iteration/sample$sample"
+        mkdir -p $benchmark_results_dir
+
+        # Generate the linpack.input dirs (in /var/lib/pbench-agent/tmp)
+        store_and_run "$benchmark_results_dir" "'$pbench_bin/bench-scripts/postprocess/linpack-prepare-input-file' --threads '$thread' --run-samples 1 --linpack-binary '$linpack_dir/$linpack_cmd'" "$benchmark-preprocess.cmd"
+        mkdir -p $benchmark_results_dir/clients/localhost
+        cp /var/lib/pbench-agent/tmp/linpack* "$benchmark_results_dir/clients/localhost"
+
+        ## Run the benchmark and start/stop perf analysis tools
+        pbench-start-tools --group=$tool_group --dir=$benchmark_results_dir
+
+        pushd "$benchmark_results_dir/clients/localhost/" >/dev/null
+        if [ "$xxx_fake_results" == "true" ]; then
+            # Report fake results instead of running the linpack command
+            record_fake_results "$benchmark_results_dir/clients/localhost/" "$thread"
+        else
+            ./linpack.sh
+        fi
+        popd >/dev/null
+
+        pbench-stop-tools --group=$tool_group --dir=$benchmark_results_dir
+        pbench-send-tools --group=$tool_group --dir=$benchmark_results_dir
+        pbench-postprocess-tools --group=$tool_group --dir=$benchmark_results_dir
+        postprocess-iteration "$benchmark_results_dir" "$benchmark_run_dir"
```

Comment:
I think we want to follow the same convention from other bench-scripts here, see https://github.com/distributed-system-analysis/pbench/blob/main/agent/bench-scripts/pbench-uperf#L772.

Reply:
I _think_ Lukas's code _is_ doing that...he cleverly wrapped it up inside a couple of utility functions.

Reply:
Well I dislike repeating my code so I used the `store_and_run` function. I think it'd actually make sense to move it to `base` but I'm too fresh to this project to consider something like that :-) Anyway is the current approach fine or do you want me to adjust it somehow?

Reply:
I, too, think making it generally available would be good.  (It would be nice to get all of the benchmark scripts consistent, where we can.)

Reply:
... in another PR.  =)